.github/workflows/optimize-ci-build-reusable.yml

```diff
@@ -0,0 +1,102 @@
+
+defaults:
+  run:
+    # use bash shell by default to ensure pipefail behavior is the default
+    # see https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#exit-codes-and-error-action-preference
+    shell: bash
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      # Setup: checkout branch
+      - name: Checkout '${{ inputs.branch }}' branch
+        uses: actions/checkout@v4
+        with:
+          ref: refs/heads/${{ inputs.branch }}
+          fetch-depth: 0 # fetches all history for all branches and tags
+
+      #########################################################################
+      # Setup: define env variables
+      - name: Set GitHub environment variables
+        run: |
+          GIT_COMMIT_HASH=$(git rev-parse ${{ inputs.branch }})
+          echo "GIT_COMMIT_HASH=$GIT_COMMIT_HASH" >> $GITHUB_ENV
+          echo "IMAGE_TAG=registry.camunda.cloud/team-camunda/optimize:pr-$GIT_COMMIT_HASH" >> $GITHUB_ENV
+          if [ "${{ inputs.previewEnv }}" = "true" ]; then
+            echo "IMAGE_TAG=registry.camunda.cloud/team-camunda/optimize:pr-$GIT_COMMIT_HASH" >> $GITHUB_ENV
+          else
+            echo "IMAGE_TAG=registry.camunda.cloud/team-optimize/optimize:pr-$GIT_COMMIT_HASH" >> $GITHUB_ENV
+          fi
+
+      #########################################################################
+      # Setup: read java version from pom
+      - name: "Read Java / Version Info"
+        id: pom-info
+        uses: YunaBraska/java-info-action@72dd3fb638d774fb4353d99d2bb3fddddd084789 # 2.1.0
+
+      #########################################################################
+      # Docker login and image tagging
+      - name: Login to Harbor registry
+        uses: ./.github/actions/login-registry
+        with:
+          secrets: ${{ toJSON(secrets) }}
+
+      #########################################################################
+      # Setup and configure Maven
+      - name: Setup Maven
+        uses: ./.github/actions/setup-maven
+        with:
+          secrets: ${{ toJSON(secrets) }}
+
+      #########################################################################
+      # Build Optimize
+      - name: Generate production .tar.gz
+        uses: ./.github/actions/run-maven
+        with:
+          parameters: -f optimize/pom.xml install -DskipTests -Dskip.docker -PrunAssembly
+
+      #########################################################################
+      # Build Docker image
+      - name: Build and push Docker image
+        uses: docker/build-push-action@ca052bb54ab0790a636c9b5f226502c73d547a25 # v5
+        with:
+          context: .
+          file: optimize.Dockerfile
+          provenance: false
+          build-args: |
+            VERSION=${{ steps.pom-info.outputs.project_version }}
+            REVISION=${{ env.GIT_COMMIT_HASH }}
+          tags: ${{ env.IMAGE_TAG }}
+          push: true
```

Comment:
Should be taken from `inputs.pushDocker` or even better from `inputs.previewEnv` since otherwise this workflow does not need to push anything.

Reply:
Yeah you are right! I think it would be better to do it through the` inputs.pushDocker` variable as in my understanding the optimize-ci workflow was also [pushing the image](https://github.com/camunda/camunda/pull/20885/files#diff-9232e03df9f79b5a046a26e4ca02fa62410d4a501a9b79be5b1f577358ef3f4dL64-L99)

Reply:
> was also [pushing the image](https://github.com/camunda/camunda/pull/20885/files#diff-9232e03df9f79b5a046a26e4ca02fa62410d4a501a9b79be5b1f577358ef3f4dL64-L99)

Yea but nobody ever uses the image because it is meant to be used for the old Optimize 7 Preview Environments in the old `camunda/camunda-optimize` hence my reference to being "unused".

Reply:
Exactly! I now understand what you meant previously :)